controllers/certificate/certificate_secret.go

```diff
@@ -45,6 +46,15 @@ func InitialiseCertificateSecret(ctx context.Context, client client.Client, log
 				ObjectMeta: metav1.ObjectMeta{
 					Namespace: secretNamespace,
 					Name:      secretName,
+					Labels: map[string]string{
+						"app.kubernetes.io/component": "api-gateway-operator.kyma-project.io",
+						"app.kubernetes.io/instance":  "api-gateway-operator-default",
+						"app.kubernetes.io/name":      "api-gateway-operator",
+						"app.kubernetes.io/part-of":   "api-gateway",
+						"app.kubernetes.io/version":   version.GetModuleVersion(),
+						"control-plane":               "controller-manager",
+						"kyma-project.io/module":      "api-gateway",
+					},
```

Comment:
I think that having only one label here specific for api-gateway would be 100% enough. With this current approach resource would have to have all the labels defined to be included in watch cache. I think having a single label that is tightly connected to api-gateway would benefit more in the long run.
```
"kyma-project.io/module": "api-gateway"
```